### PR TITLE
Funqy HTTP query parameter binding

### PR DESCRIPTION
**Pull request by patriot1burke**
For GET requests, query parameters can now be mapped to the input parameter.  Previously, GET requests could not send input to functions.

@gunnarmorling

### CODE REVIEW

File: docs/src/main/asciidoc/funqy-http.adoc [hunk 1]
```diff
@@ -97,15 +107,198 @@ curl "http://localhost:8080/double" \
...
-    <groupId>io.quarkus</groupId>
-    <artifactId>quarkus-funqy-http</artifactId>
-</dependency>
+@Funq
+public String hello(Map<String, Integer> map) {
+...
+}
+----
+
+Key values must be a primitive type or `String`.  Values can be primitives, `String`, `OffsetDateTime` or a complex
```
Comment (loicmathieu) [inline docs/src/main/asciidoc/funqy-http.adoc:125]: Why `OffsetDateTime` and not `Date` of the other date types inside java.time ?

File: docs/src/main/asciidoc/funqy-http.adoc [hunk 2]
```diff
@@ -97,15 +107,198 @@ curl "http://localhost:8080/double" \
...
+    }
+}
+
+----
+
+To invoke a GET request, just list the `pets` query parameter multiple times.
+
+[source, shell]
+----
+curl "http://localhost:8080/greet?pets=itchy&pets.scratchy"
```
Comment (loicmathieu) [inline docs/src/main/asciidoc/funqy-http.adoc:221]:
```suggestion
curl "http://localhost:8080/greet?pets=itchy&pets=scratchy"
```

File: extensions/funqy/funqy-server-common/runtime/src/main/java/io/quarkus/funqy/runtime/query/QueryObjectMapper.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        if (clz.equals(byte.class) || clz.equals(Byte.class)) {
+            return (strVal) -> {
+                return Byte.valueOf(strVal);
+            };
+        }
+        if (clz.equals(OffsetDateTime.class)) {
+            return (strVal) -> {
+                return OffsetDateTime.parse(strVal);
+            };
+        }
```
Comment (loicmathieu) [inline extensions/funqy/funqy-server-common/runtime/src/main/java/io/quarkus/funqy/runtime/query/QueryObjectMapper.java:67]: I didn't saw any test with `OffsetDateTime`.

**Review by loicmathieu — DISMISSED** [review]
LGTM.

I added a question about `OffsetDateTime` I also think more documentation of date support (which format for example) needs to be written.

**Comment by gsmet** [comment]
@patriot1burke could you have a look at the review? I'm releasing CR1 tomorrow. Thanks!

File: docs/src/main/asciidoc/funqy-http.adoc [hunk 4]
```diff
@@ -97,15 +107,198 @@ curl "http://localhost:8080/double" \
...
+
+public class MyFunctions {
+    @Funq
+    public String greet(Person p) {
+       return "Hello " + p.getFirst() + " " + p.getLast();
+    }
+}
+----
+
+Property values can be any primitive type except `char`, `String`, and `OffsetDateTime`.
```
Comment (gunnarmorling) [inline docs/src/main/asciidoc/funqy-http.adoc:157]: You don't rule out `char` in your list above?

**Review by gunnarmorling — COMMENTED** [review]
Didn't look closely at the impl, but from the description in the guide it looks very useful 👍

File: docs/src/main/asciidoc/funqy-http.adoc (hunk 1 above)
Comment (patriot1burke) [reply docs/src/main/asciidoc/funqy-http.adoc:125]: Its parse() method is a specific format ISO-8601 where Date requires a DateFormat.  Using offset for date time queries is also a better data structure as dates are usually stored by absolute time.

File: extensions/funqy/funqy-server-common/runtime/src/main/java/io/quarkus/funqy/runtime/query/QueryObjectMapper.java (hunk 3 above)
Comment (patriot1burke) [reply extensions/funqy/funqy-server-common/runtime/src/main/java/io/quarkus/funqy/runtime/query/QueryObjectMapper.java:67]: test added

**Comment by patriot1burke** [comment]
@gsmet Resovled all comments.  Should be good.

**Review by gsmet — APPROVED** [review]
Loïc looked OK with it apart from the comments so merging.

**Pull request merged**